testing/e2e/suites/operatorConfig.go

```diff
@@ -39,11 +39,7 @@ func (c cfgBuilder) withOsm(in infra.Provisioned, enabled ...bool) cfgBuilderWit
 	for _, e := range enabled {
 		// osm tests can only work if the cluster has osm installed.
 		// filter out any enabled on clusters without osm
-		if !osmCluster && e {
-			continue
-		}
-
-		if osmCluster && !e {
+		if osmCluster != e {
```

Comment:
this revert isn't correct. merge with main

Reply:
reverted - but when is one ever not equal to the other? does it somehow involve a nil case?